pakyow-support/lib/pakyow/support/defineable.rb

```diff
@@ -103,32 +156,67 @@ def define(&block)
     class State
       using DeepDup
 
-      attr_reader :name, :object, :instances
+      attr_reader :name, :object, :instances, :priorities
+
+      PRIORITIES = { default: 0, high: 1, low: -1 }.freeze
 
       def initialize(name, object)
         @name = name.to_sym
         @object = object
         @instances = []
+        @priorities = {}
       end
 
       def initialize_copy(original)
         super
         @instances = original.instances.deep_dup
       end
 
+      # TODO: we handle both instances and classes, so reconsider the variable naming
       def <<(instance)
-        unless instance.class.ancestors.include?(object)
+        register(instance)
+      end
+
+      # TODO: we handle both instances and classes, so reconsider the variable naming
+      def register(instance, priority: :default)
+        unless priority.is_a?(Integer)
+          priority = PRIORITIES.fetch(priority) {
+            raise ArgumentError, "Unknown priority `#{priority}'"
+          }
+        end
+
+        ancestors = if instance.respond_to?(:new)
+          instance.ancestors
+        else
+          instance.class.ancestors
+        end
+
+        unless ancestors.include?(object)
           raise ArgumentError, "Expected instance of '#{object}'"
         end
 
         instances << instance
+
+        priorities[instance] = priority
+        reprioritize!
       end
 
       def freeze
         instances.each(&:freeze)
         instances.freeze
         super
       end
+
+      def reset
+        object.reset
+        @instances = []
+      end
+
+      def reprioritize!
+        @instances.sort! { |a, b|
```

Comment:
Ah, router priority isn't defined! I'll add it.

The case that led to router priority is when a router defines a wildcard route. You usually want the wildcard to be the last route matched, because it's usually some sort of fallback. Using router priority, you can designate the router containing the wildcard as low priority.